buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -248,7 +250,20 @@ public static String gitRevision(File rootDir) {
             }
             final String ref = readFirstLine(head);
             if (ref.startsWith("ref:")) {
-                revision = readFirstLine(gitDir.resolve(ref.substring("ref:".length()).trim()));
+                String branchname = ref.substring("ref:".length()).trim();
```

Comment:
Sorry for being pedantic but this threw me off when I was digging through the git file structure. This is actually the name of the reference (ex: `refs/heads/master`) not simply the name of the branch. Can we rename this `refName` or similar to account for this?

Reply:
That's the good kind of pedantic; you're absolutely right. I'll do it.